pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -37,7 +37,12 @@
 class CheckpointConnector:
     def __init__(self, trainer: "pl.Trainer", resume_from_checkpoint: Optional[_PATH] = None) -> None:
         self.trainer = trainer
-        self.resume_checkpoint_path = resume_from_checkpoint
+        self.resume_checkpoint_path: Optional[_PATH] = None
```

Comment:
Do we need to keep this attribute?

Reply:
I think so, it helps CheckpointConnector remember which path `_loaded_checkpoint` is from, instead of passing a path to every function and loading path file contents each time. wdyt?

Reply:
If we deprecate the following: https://github.com/PyTorchLightning/pytorch-lightning/pull/9693/files#r716905724
then I think we can remove it. The only thing that's not issue is keeping `rank_zero_info` message in `resume_end`

Reply:
aside from losing the message, would it be odd keeping `_loaded_checkpoint` without `resume_checkpoint_path` (where is `_loaded_checkpoint` from?)

Reply:
I wouldn't say so

Any preference @ananthsub or @awaelchli?

Reply:
I recommend not keeping the loaded checkpoint in memory beyond the resuming procedure. Please keep the log message acknowledging loading of the checkpoint and the path to it.